create.php

```diff
@@ -0,0 +1,56 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
```

Comment:
Like the create_entry form object, this whole file should not be required. The edit.php should be able to easily handle both create and edit and is how almost all forms in moodle work. 

I'll double confused because this create.php file doesn't seem to be linked to from any buttons so its not actually used anywhere?

Broadly the rest of this pull request is great - but its also getting a but larger than ideal and it's including other changes which should be independent